tools/glide-pindeps/pindeps.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/openshift/origin/tools/glide-pindeps/glide"
+)
+
+func readLockFile(filepath string) (*glide.LockFile, error) {
+	b, err := ioutil.ReadFile(filepath)
+	if err != nil {
+		return nil, err
+	}
+
+	glidelock := &glide.LockFile{}
+	glidelock.Decode(b)
+
+	return glidelock, err
+}
+
+func readYamlFile(filepath string) (*glide.YamlFile, error) {
+	b, err := ioutil.ReadFile(filepath)
+	if err != nil {
+		return nil, err
+	}
+
+	glideyaml := &glide.YamlFile{}
+	glideyaml.Decode(b)
+
+	return glideyaml, err
+}
+
+func pinDependencies(lockFilePath, yamlFilePath string) error {
+	if len(lockFilePath) == 0 {
+		return fmt.Errorf("a glide.lock file must be specified via the --glide.lock flag\n")
+	}
+	if len(yamlFilePath) == 0 {
+		return fmt.Errorf("a glide.yaml file must be specified via the --glide.yaml flag\n")
+	}
+
+	lockfile, err := readLockFile(lockFilePath)
+	if err != nil {
+		return fmt.Errorf("error: unable to read glide.lock: %v\n", err)
+	}
+
+	yamlfile, err := readYamlFile(yamlFilePath)
+	if err != nil {
+		return fmt.Errorf("error: unable to read glide.yaml: %v\n", err)
+	}
+
+	missing, err := glide.MissingImports(lockfile, yamlfile)
+	if err != nil {
+		return fmt.Errorf("error: unable to calculate missing imports: %v\n", err)
+	}
+
+	b, err := yamlfile.AppendAndEncodeRaw(missing)
+	if err != nil {
+		return fmt.Errorf("error: unable to encode updated glide.yaml data: %v\n", err)
+	}
+
+	fmt.Fprintf(os.Stdout, "%v", string(b))
+	return nil
+}
+
+func main() {
+	var lockFilePath string
+	var yamlFilePath string
+
+	flag.Usage = func() {
+		usageText := `This program updates all non-forked dependencies in glide.yaml to use the SHAs specified in glide.lock.
+Any existing dependencies on a glide.yaml file containing a "repo" field are ignored.
+`
+		fmt.Fprintf(os.Stderr, "%s\n", usageText)
+		flag.PrintDefaults()
+	}
+	flag.StringVar(&lockFilePath, "glide.lock", "glide.lock", "Path to glide.lock file used to indicate which dependencies should be updated or pinned to a specific revision in the glide.yaml file")
+	flag.StringVar(&yamlFilePath, "glide.yaml", "glide.yaml", "Path to glide.yaml file used to pin dependencies from the given --glide.lock file")
+	flag.Parse()
+
+	if err := pinDependencies(lockFilePath, yamlFilePath); err != nil {
```

Comment:
Much better.  Why not put your work in a different package hanging on an options struct so someone else could depend on it?  Not that I have a use-case, but it's a small thing to make the eco-system a little better.